pkg/controller/account/account_controller.go

```diff
@@ -448,47 +507,57 @@ func (r *ReconcileAccount) BuildAccount(reqLogger logr.Logger, awsClient awsclie
 	return *orgOutput.CreateAccountStatus.AccountId, nil
 }
 
-// BuildUser takea all parameters required to create a user, user secret
-func (r *ReconcileAccount) BuildUser(reqLogger logr.Logger, awsClient awsclient.Client, account *awsv1alpha1.Account, nameSpace string) (string, error) {
+// BuildIAMUser takes all parameters required to create a user, user secret
+func (r *ReconcileAccount) BuildIAMUser(reqLogger logr.Logger, awsClient awsclient.Client, account *awsv1alpha1.Account, iamUserName string, nameSpace string) (string, error) {
 	reqLogger.Info("IAM User Created")
 
-	// create iam user
+	// Create IAM user
 	_, userErr := CreateIAMUser(awsClient, iamUserName)
 	// TODO: better error handling but for now scrap account
 	if userErr != nil {
-		setAccountStatus(reqLogger, account, "Failed to create account", awsv1alpha1.AccountFailed, "Failed")
+		failedToCreateIAMUserMsg := fmt.Sprintf("Failed to create IAM user %s", iamUserName)
+		setAccountStatus(reqLogger, account, failedToCreateIAMUserMsg, awsv1alpha1.AccountFailed, "Failed")
 		err := r.Client.Status().Update(context.TODO(), account)
 		if err != nil {
 			return "", err
 		}
-		reqLogger.Info("Failed to create user")
+		reqLogger.Info(failedToCreateIAMUserMsg)
 		return "", userErr
 	}
 
-	reqLogger.Info("Attaching Admin Policy to IAM user")
+	reqLogger.Info(fmt.Sprintf("Attaching Admin Policy to IAM user %s", iamUserName))
 	// Setting user access policy
 	_, policyErr := AttachAdminUserPolicy(awsClient, iamUserName)
 	if policyErr != nil {
-		setAccountStatus(reqLogger, account, "Failed to set admin policy", awsv1alpha1.AccountFailed, "Failed")
-		r.setStatusFailed(reqLogger, account, "Failed to build user")
+		failedToAttachUserPolicyMsg := fmt.Sprintf("Failed to attach admin policy to IAM user %s", iamUserName)
```

Comment:
Agree but these strings are not constant since i'm including the IAM user name. I think we should create a separate card to clean up errors for this operator.

Reply:
:+1: